spring-core/src/test/kotlin/org/springframework/core/KotlinReflectionParameterNameDiscovererTests.kt

```diff
@@ -42,11 +42,22 @@ class KotlinReflectionParameterNameDiscovererTests {
 		assertThat(actualParams, `is`(arrayOf("message")))
 	}
 
+	@Test
+	fun getParameterNamesOnExtensionMethod() {
+		val method = ReflectionUtils.findMethod(UtilityClass::class.java, "identity", String::class.java)!!
+		val actualParams = parameterNameDiscoverer.getParameterNames(method)!!
+		assertThat(actualParams, `is`(arrayOf("\$receiver")))
```

Comment:
As described in my JIRA issue comment, I think we should check the actual parameter name against the one retrieved from `Method`.